### PR TITLE
Add admin role check

### PR DESCRIPTION
**Pull request by eanlain**

### CODE REVIEW

File: app/models/storytime/role.rb [hunk 1]
```diff
@@ -6,6 +6,10 @@ class Role < ActiveRecord::Base
 
     validates :name, uniqueness: true
 
+    def admin?
+      name == "admin"
+    end
+
```
Comment (bcroesch) [inline app/models/storytime/role.rb]: I should have said this to start, but we should probably have these for all of our roles (editor, writer, etc.)

**Comment by eanlain** [comment]
Added additional role check

**Comment by dvanderbeek** [comment]
Should we change this so it loops through the roles in the db...doing it this way creates a dependency between the code and specific role names being in the database

**Comment by bcroesch** [comment]
Honestly id like to see us get the role stuff out of the db entirely and
into an enum. Would clean a lot of this up and also make our seeding a
little cleaner and easier.

**Comment by dvanderbeek** [comment]
hmm. I guess that depends on whether we ever want anyone to be able to add their own roles. if there’s a role/action combination that you can’t do with the three roles we came up with, I could see wanting to add a new role.

—

**Comment by bcroesch** [comment]
Yeah, IIRC that's why we decided to put them in the db in the first place and is a valid concern. Part of me wonders if the config/initializer isn't a better place to define additional roles, so that they are present in version control and are consistent across envs. 

Either way, if we end up having people add roles through the dashboard eventually, I'd be nervous about looping through all the roles in the db and defining a method based on the db content. That would be a pretty huge potential security hole where someone could put something malicious in the role name & have it get executed as code.

**Comment by dvanderbeek** [comment]
True. The initializer could work.



—

**Pull request merged**